contracts/src/ethereum/TokenBridge.sol

```diff
@@ -140,27 +144,48 @@ contract TokenBridge is
         l1Token.transfer(recipient, amount);
     }
 
-     function consumeBridgeRewardMessage(address l1Token, address recipient, uint256 amount) internal {
-        emit LogBridgeReward(l1Token, recipient, amount);
+     function consumeBridgeRewardMessage(address recipient, uint256 amount) internal {
+        emit LogBridgeReward(recipient, amount);
 
-        uint256[] memory payload = new uint256[](5);
+        uint256[] memory payload = new uint256[](4);
         payload[0] = BRIDGE_REWARD_MESSAGE;
-        payload[1] = uint256(l1Token);
-        payload[2] = uint256(recipient);
-        payload[3] = amount & (UINT256_PART_SIZE - 1);
-        payload[4] = amount >> UINT256_PART_SIZE_BITS;
+        payload[1] = uint256(recipient);
+        payload[2] = amount & (UINT256_PART_SIZE - 1);
+        payload[3] = amount >> UINT256_PART_SIZE_BITS;
 
         messagingContract.consumeMessageFromL2(l2TokenBridge, payload);
     }
 
-    function receiveRewards(address l1Token, address recipient, uint256 amount) isApprovedToken(l1Token) external {
-        consumeBridgeRewardMessage(l1Token, recipient, amount);
+    function receiveRewards(address recipient, uint256 amount) external {
+        consumeBridgeRewardMessage(recipient, amount);
         require(recipient != address(0x0), "INVALID_RECIPIENT");
 
-        IStaticATokenLM staticAToken =  IStaticATokenLM(l1Token);
-        staticAToken.claimRewardsToSelf(true);
+        require(approvedL1Tokens.length > 1, "REWARD TOKEN NOT FOUND");
+        IERC20 rewardToken = IStaticATokenLM(approvedL1Tokens[0]).REWARD_TOKEN();
```

Comment:
We should actually make sure that the reward token is aave instead of issuing any token here

Reply:
In that case should we even check if its aave? Should we just the address of aave token, would save gas as well